### ISSUE TITLE
Observations regarding current MES search behavior

### ISSUE THREAD

**Issue by patrickab** [patrickab/Maximum-SAIL#2]
As mentioned in issue #1, MES has shown capable of producing remarkable results, However, some aspects still apear strange to me. First of all, MES seems to evaluate a lot around high performing regions and within the xfoil-domain hardly reaches objective values above 5.0. This happens for a very long time, until suddenly, the algorithm finds one high performing solution - in the most extreme case until now of up to 5.7. However, after this happens, the algorithm struggles to find high performing solutions again.

For future improvements, I have two promising easy-to-implement solutions on my mind

1. Use a [Evolutionary Strategy Emitter](https://docs.pyribs.org/en/stable/api/ribs.emitters.EvolutionStrategyEmitter.html), in order to converge towards high-performing acquisition values more rapidly. This could potentially drive down the required Max Entropy evaluations, making gradient based strategies or covariance based strategies like [CMA-ES](https://arxiv.org/pdf/1604.00772.pdf) a feasible solution. 

2. Maybe, finding a solution to integrate knowledge about the current highest performing objective elites into the acquisition elites, might be a good idea. Potentially crossover-breeding of the acquistion archive produced inside the [eval_xfoil_loop](https://github.com/patrickab/Maximum-SAIL/blob/ad52567cb64f2206f54068d7e0b0efbb8ede0aa9/sail_xfoil/xfoil/eval_xfoil_loop.py) could be a solution, although I still have no concrete idea.

**Comment by sumny** [comment]
Is it possible that due to the mutation of the "high-performing" candidates as described in #1 certain behavior spaces are much more focused on in early iterations of the MES SAIL Algorithm (i.e., suppose we have one niche where high MES values can be found; then we generate many more mutants that also fall into this niche (?)) - probably this can also hinder early performance because we evaluate too many candidate points close to each other in behavior space which somewhat translates also into x space. It might be better to generate many diverse candidate points (with respect to the behavior functions and niches) especially in early stages - also note that MES value ranges of different niches might not be comparable and performing a mutation of "high performing" candidates based on a threshold of the MES value can introduce a bias to only search in certain niches. This might explain why MES SAIL still manages to find good solutions at the very end because it then managed to explore the search space. It might be beneficial to ensure that already at early stages only one candidate per niche can be evaluated and maintain high diversity of candidates with respect to their behavior function values from the very beginning.

**Comment by patrickab** [comment]
Indeed, thats exactly what's happening at the moment.  Actually, more than that, only one region will be focused by, as shown in this [video](https://github.com/patrickab/Maximum-SAIL/blob/master/sail_xfoil/documentation_references/acq_map_92fa814.mp4). In this benchmark i rendered the acquisition archive once before, and once after this routine:

```
            if remaining_evals % (n_evals//5) == 0:
                if acq_flag and self.acq_mes_flag:
                    self.visualize_archive(target_archive, acq_flag=True)

                    # update elites with high acquisition values
                    target_elites = target_archive.as_pandas(include_solutions=True)
                    update_elites = target_elites[target_elites['objective'] > ACQ_MES_MIN_THRESHHOLD*1.5]
                    self.update_archive(candidate_sol=update_elites.solution_batch(), candidate_bhv=update_elites.measures_batch(), acq_flag=True)

                    # Boost highestperforming elites
                    percentages = [0.4, 0.2, 0.1, 0.05]
                    for percentage in percentages:
                        self.update_seed()
                        target_elites = target_archive.as_pandas(include_solutions=True).sort_values(by='objective', ascending=False)
                        update_elites = target_elites.head(int(percentage*target_archive.stats.num_elites))
                        self.update_archive(candidate_sol=update_elites.solution_batch(), candidate_bhv=update_elites.measures_batch(), acq_flag=True)

                    if target_archive.stats.num_elites > 80:
                        # remove elites with low acquisition values
                        target_elites = target_archive.as_pandas(include_solutions=True).sort_values(by='objective', ascending=False)
                        target_elites = target_elites[target_elites['objective'] > 1.5*ACQ_MES_MIN_THRESHHOLD]
                        target_archive.clear()
                        target_archive.add(solution_batch=target_elites.solution_batch(), objective_batch=target_elites.objective_batch(), measures_batch=target_elites.measures_batch())

                    print("Remaining Evaluations: ", remaining_evals)
                    self.visualize_archive(target_archive, acq_flag=True)
```
